subsys/nrf_security/src/ssf_secdom/bounce_buffers.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+
+#include "bounce_buffers.h"
+
+/* k_heap_alloc allocated memory is aligned on a multiple of pointer sizes. The HW's DataUnit size
+ * must match this Zephyr behaviour.
+ */
+BUILD_ASSERT(CACHE_DATA_UNIT_SIZE == sizeof(uintptr_t));
+
+static K_HEAP_DEFINE(out_buffer_heap,
+		     ROUND_UP(CONFIG_PSA_SSF_CRYPTO_CLIENT_OUT_HEAP_SIZE, CACHE_DATA_UNIT_SIZE));
+
+void *bounce_buffers_prepare(void *original_buffer, size_t size)
+{
+	void *out_buffer = NULL;
+
+	if ((IS_ALIGNED(original_buffer, CACHE_DATA_UNIT_SIZE)) &&
+	    (IS_ALIGNED(size, CACHE_DATA_UNIT_SIZE))) {
+		out_buffer = original_buffer;
```

Comment:
should we check if `original_buffer` is NULL here?

Reply:
Yea it might be best to just pass through null pointers and leave the dirty work the psa implementation.

Also you do need to check for 0 size, @magnev  just fixed that!
https://github.com/nrfconnect/sdk-nrf/pull/21354

Reply:
I looked at what happens if original_buffer is NULL.

There are two code paths.

In the first code path NULL is returned, and then the calling function returns PSA_ERROR_INSUFFICIENT_MEMORY.
I suppose that is reasonable.

In the second code path a memcpy will read from original_buffer, at which point IRONside should give us a memory access error report. Which is more useful than any kind of error handling I can think of.

IMHO this function shouldn't have special handling for NULL.